ci/lua-types.py

```diff
@@ -0,0 +1,2559 @@
+import re
+import xml.etree.ElementTree as ET
+from collections.abc import Iterable
+from dataclasses import dataclass
+from pathlib import Path
+from xmlrpc.client import boolean
+
+PATH_XML = "./library/xml"
+PATH_OUTPUT = "./types/library/"
+PATH_LIB_CONFIG = "./types/config.json"
+PATH_CONFIG = ".luarc.json"
+
+ALIAS = "---@meta\n\n---@alias bitfield integer\n"
+LIB_CONFIG = """{
+  "name": "DFHack Lua",
+  "words": [
+      "dfhack"
+  ]
+}\n"""
+
+CONFIG = f"""{{
+  "workspace.library": ["{PATH_OUTPUT}"],
+  "workspace.ignoreDir": ["build"]
+}}\n"""
+
+DF_GLOBAL = [
```

Comment:
these lists will get out of date. can they be parsed from the XML or generated headers?

Reply:
I parsed these lists traversing `df` and `df.globals` by lua script in game. If there is another static option instead, it would be better to use it. Give me a hint:)

Reply:
For the best result we should somehow reimplement of exposing login in `LuaWrapper.cpp`

Reply:
for the globals, you can get them from the xml:
```sh
fgrep '<global-object' library/xml/df.globals.xml | sed -r "s/.* name='([^']+).*/\1/" | sort
```
that's shell, but I'm sure you can translate that to python.